### ISSUE TITLE
Broken source links for ruby-marshal package

### ISSUE THREAD

**Issue by filib** [haskell/haddock#426]
Firstly, thank you for all your hard work on Haddock. It is really appreciated. 

## Overview

When I try to build the haddocks for my package [ruby-marshal](https://github.com/filib/ruby-marshal), the source links are broken (*please note, I'm not referring to links embedded in the source but links to the source*). I've tried using Haddock 2.16.0 and 2.16.1 but both generate broken source links, i.e. the source HTML files do not appear to be created at all.

When I try to do the same thing using GHC 7.8.4, Haddock 2.14.3, and cabal-install 1.22.6.0, the source links are not broken.

## Reproduction Steps

First clone the ruby-marshal repo, check out the latest version and build the package.

``` 
% git clone git@github.com:filib/ruby-marshal.git
% cd ruby-marshal
% git checkout 0.1.0
% cabal sandbox init
% cabal install
% cabal configure
% cabal haddock --hoogle --hyperlink-source
```

Then open the haddocks located at `dist/doc/html/ruby-marshal/index.html`, click on `Data.Ruby.Marshal` and finally click on `source`.

## Program Output

```
% cabal haddock --hoogle --hyperlink-source
Running Haddock for ruby-marshal-0.1.0...
Warning: The program 'hscolour' version >=1.8 is required but it could not be
found.
Preprocessing library ruby-marshal-0.1.0...
Warning: The documentation for the following packages are not installed. No
links will be generated to these packages: cereal-0.4.1.1, mtl-2.2.1,
primitive-0.6, string-conv-0.1, text-1.2.1.3, vector-0.10.12.3
Haddock coverage:
Warning: Couldn't find .haddock for export getWord8
Warning: Couldn't find .haddock for export getWord16le
Warning: Couldn't find .haddock for export getWord32le
  77% ( 10 / 13) in 'Data.Ruby.Marshal.Int'
  Missing documentation for:
    getWord8
    getWord16le
    getWord32le
 100% (  5 /  5) in 'Data.Ruby.Marshal.Encoding'
  19% (  3 / 16) in 'Data.Ruby.Marshal.RubyObject'
  Missing documentation for:
     (src/Data/Ruby/Marshal/RubyObject.hs:64)
     (src/Data/Ruby/Marshal/RubyObject.hs:68)
     (src/Data/Ruby/Marshal/RubyObject.hs:74)
     (src/Data/Ruby/Marshal/RubyObject.hs:80)
     (src/Data/Ruby/Marshal/RubyObject.hs:86)
     (src/Data/Ruby/Marshal/RubyObject.hs:92)
     (src/Data/Ruby/Marshal/RubyObject.hs:98)
     (src/Data/Ruby/Marshal/RubyObject.hs:104)
     (src/Data/Ruby/Marshal/RubyObject.hs:110)
     (src/Data/Ruby/Marshal/RubyObject.hs:118)
     (src/Data/Ruby/Marshal/RubyObject.hs:129)
     (src/Data/Ruby/Marshal/RubyObject.hs:135)
     (src/Data/Ruby/Marshal/RubyObject.hs:139)
 100% (  9 /  9) in 'Data.Ruby.Marshal.Monad'
 100% ( 22 / 22) in 'Data.Ruby.Marshal.Types'
 100% (  4 /  4) in 'Data.Ruby.Marshal.Get'
 100% (  9 /  9) in 'Data.Ruby.Marshal'
Warning: Data.Ruby.Marshal.Int: could not find link destinations for:
    Get Serialize
Warning: Data.Ruby.Marshal.RubyObject: could not find link destinations for:
    Vector
Warning: Data.Ruby.Marshal.Monad: could not find link destinations for:
    Get MonadState Vector
Warning: Data.Ruby.Marshal.Types: could not find link destinations for:
    MonadState Vector
Documentation created: dist/doc/html/ruby-marshal/index.html,
dist/doc/html/ruby-marshal/ruby-marshal.txt
```

### GHC Version

```
% ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.10.1
```

### Cabal Version

```
% cabal --version
cabal-install version 1.22.2.0
using version 1.22.2.0 of the Cabal library
```

### Haddock Version

```
% haddock --version
Haddock version 2.16.1, (c) Simon Marlow 2006
Ported to use the GHC API by David Waern 2006-2008
```

### Platform | OS

```
Software:

    System Software Overview:

      System Version: OS X 10.10.4 (14E46)
      Kernel Version: Darwin 14.4.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: Philip’s MacBook Air
      User Name: Philip Cunningham (filib)
      Secure Virtual Memory: Enabled
      Time since boot: 3 days 16:22
```

<hr/>
https://github.com/filib/ruby-marshal/issues/50

**Comment by filib** [comment]
This appears to have been fixed by updating `cabal-install` to 1.22.6.0. Apologies for any inconvenience and, again, thank you for all your hard work on Haddock!

**Comment by Fuuzetsu** [comment]
We're moving away from hscolour next release but ensuring a recent cabal-install is always worthwhile. Glad it works for you
